packages/event-handler/tests/unit/rest/middleware/cors.test.ts

```diff
@@ -0,0 +1,402 @@
+      apiId: 'test-api',
+      protocol: 'HTTP/1.1',
+      httpMethod: method,
+      path: '/test',
+      stage: 'test',
+      requestId: 'test-request-id',
+      requestTime: '09/Apr/2015:12:34:56 +0000',
+      requestTimeEpoch: 1428582896000,
+      authorizer: {},
+      identity: {
+        accessKey: null,
+        accountId: null,
+        apiKey: null,
+        apiKeyId: null,
+        caller: null,
+        clientCert: null,
+        cognitoAuthenticationProvider: null,
+        cognitoAuthenticationType: null,
+        cognitoIdentityId: null,
+        cognitoIdentityPoolId: null,
+        principalOrgId: null,
+        sourceIp: '127.0.0.1',
+        user: null,
+        userAgent: 'Custom User Agent String',
+        userArn: null,
+      },
+      resourceId: 'test-resource',
+      resourcePath: '/test',
+    },
+  };
+}
+
+function createMockRequestContext(
+  method: string,
+  origin?: string
+): RequestContext {
+  const headers = new Headers();
+  if (origin) {
+    headers.set('Origin', origin);
+  }
+
+  const request = new Request('https://example.com/test', {
+    method,
+    headers,
+  });
+
+  const response = new Response('', { status: 200 });
+
+  return {
+    request,
+    res: response,
+    event: createMockAPIGatewayEvent(method, origin ? { Origin: origin } : {}),
+    context: mockContext,
+  };
+}
+
+describe('CORS Configuration', () => {
+  it('should use default configuration when no options provided', async () => {
+    const middleware = cors();
+    const reqCtx = createMockRequestContext('GET');
+    const next = vi.fn().mockResolvedValue(undefined);
+
+    await middleware({}, reqCtx, next);
+
+    expect(reqCtx.res.headers.get('Access-Control-Allow-Origin')).toBe('*');
+    expect(next).toHaveBeenCalledTimes(1);
+  });
+
+  it('should merge user options with defaults', async () => {
+    const options: CorsOptions = {
+      origin: 'https://example.com',
+      credentials: true,
+    };
+
+    const middleware = cors(options);
+    const reqCtx = createMockRequestContext('GET');
+    const next = vi.fn().mockResolvedValue(undefined);
+
+    await middleware({}, reqCtx, next);
+
+    expect(reqCtx.res.headers.get('Access-Control-Allow-Origin')).toBe('https://example.com');
+    expect(reqCtx.res.headers.get('Access-Control-Allow-Credentials')).toBe('true');
+    expect(next).toHaveBeenCalledTimes(1);
+  });
+
+  it('should handle expose headers configuration', async () => {
+    const options: CorsOptions = {
+      exposeHeaders: ['Content-Length', 'X-Custom-Header'],
+    };
+
+    const middleware = cors(options);
+    const reqCtx = createMockRequestContext('GET');
+    const next = vi.fn().mockResolvedValue(undefined);
+
+    await middleware({}, reqCtx, next);
+
+    expect(reqCtx.res.headers.get('Access-Control-Expose-Headers')).toBe(
+      'Content-Length, X-Custom-Header'
+    );
+    expect(next).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('Origin Resolution', () => {
```

Comment:
Please don't nest or add multiple `describe` statements - just keep one at the top and all the `it` cases within at the top level.